spyder/widgets/collectionseditor.py

```diff
@@ -1819,6 +1862,10 @@ def set_data(self, data):
     def get_title(self):
         """Get model title"""
         return self.editor.source_model.title
+    
+    def close_window(self):
+        if self.parent:
+            self.parent.reject
```

Comment:
```suggestion
        if self.parent():
            self.parent().reject()
```

- Use `parent` method.
- `reject` was not being called, so this method was not having effect.